build-version.sh

```diff
@@ -57,6 +57,9 @@ done
 #
 # TODO: maybe we want to make a ~/.cargo-quickinstall/bin to untar into,
 # and add symlinks into ~/.cargo/bin, to aid debugging?
+#
+# BINARIES is a space-separated list of files, so it can't be quoted
+# shellcheck disable=SC2086
 tar --format=v7 -czf "${TEMPDIR}/${CRATE}-${VERSION}-${TARGET_ARCH}.tar.gz" $BINARIES
```

Comment:
I'm thinking perhaps we can force the highest compression rate here?

```suggestion
tar --format=v7 -c $BINARIES | gzip -9 -c > "${TEMPDIR}/${CRATE}-${VERSION}-${TARGET_ARCH}.tar.gz"
```

I'm not sure whether `tar` uses the highest compression level or not.

Reply:
I have taken the philosophy that you get multiple orders of magnitude by installing from precompiled binaries, so I haven't gone chasing after marginal compression gains.

I also wasn't sure whether the windows runners had gzip as a separate tool (I was surprised to find that `tar` is available by default on windows these days). If you think it is, I'm happy to go with this.

Reply:
> I also wasn't sure whether the windows runners had gzip as a separate tool (I was surprised to find that tar is available by default on windows these days). If you think it is, I'm happy to go with this.

I'm checking, but AFAIK `tar` simply spawns a new command for the compression/decompression for gz and other command, so it should be available.

Reply:
I didn't find `gzip` or `tar` on https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md , perhaps we can simply test it out on CI?

Reply:
oooh! we should install one of those tmux debugger actions that we used at my previous job - https://github.com/marketplace/actions/debugging-with-tmate

Reply:
The CI on windows runs just fine, so I will close this.

Reply:
> oooh! we should install one of those tmux debugger actions that we used at my previous job - https://github.com/marketplace/actions/debugging-with-tmate

I have no idea such thing exists, I debugged my GHA script by pushing new commits...

Reply:
note that this won't actually get exercised until we try to build a package that's not cancelled. We will definitely be building a package for cargo-quickinstall when we do our release, so I'm fine to merge this to main and revert it then if it causes problems.

Reply:
ooh! I am cleverer than I give myself credit for: selfbuild *does* exercise this codepath on all PRs.